server/src/main/java/com/linecorp/centraldogma/server/internal/api/AdministrativeService.java

```diff
@@ -88,45 +88,61 @@ public ServerStatus status() {
+        //                   | false -> true | setWritable(true) & start executor.
+        //-------------------+---------------+-----------------------------------
+        //    false -> false | true -> false | Stop executor.
+        //                   | false -> true | Start executor.
         final boolean writable = writableNode.asBoolean();
         final boolean replicating = replicatingNode.asBoolean();
         if (writable && !replicating) {
             return HttpApiUtil.throwResponse(ctx, HttpStatus.BAD_REQUEST,
                                              "'replicating' must be 'true' if 'writable' is 'true'.");
         }
+        if (oldStatus.writable == writable && oldStatus.replicating == replicating) {
+            throw HttpStatusException.of(HttpStatus.NOT_MODIFIED);
+        }
+
+        if (oldStatus.writable != writable) {
+            executor().setWritable(writable);
+            if (writable) {
+                logger.warn("Left read-only mode. replication: {}, replicating");
+            } else {
+                logger.warn("Entered read-only mode. replication: {}, replicating");
+            }
+        }
 
-        if (oldStatus.writable) {
-            if (!writable) { // writable -> unwritable
-                executor().setWritable(false);
-                if (replicating) {
-                    logger.warn("Entered read-only mode with replication enabled");
-                    return CompletableFuture.completedFuture(status());
-                } else {
-                    logger.warn("Entering read-only mode with replication disabled ..");
-                    return executor().stop().handle((unused, cause) -> {
-                        if (cause != null) {
-                            logger.warn("Failed to stop the command executor:", cause);
-                        } else {
-                            logger.info("Entered read-only mode with replication disabled");
-                        }
-                        return status();
-                    });
-                }
+        if (oldStatus.replicating != replicating) {
+            if (replicating) {
+                return executor().start().handle((unused, cause) -> {
+                    if (cause != null) {
+                        logger.warn("Failed to start the command executor:", cause);
+                    } else {
+                        logger.info("Enabled replication. read-only: {}", !writable);
+                    }
+                    return status();
+                });
+            } else {
+                return executor().stop().handle((unused, cause) -> {
+                    if (cause != null) {
+                        logger.warn("Failed to stop the command executor:", cause);
+                    } else {
+                        logger.info("Disabled replication");
+                    }
+                    return status();
+                });
```

Comment:
nit: we can return early.
```suggestion
            }
            return executor().stop().handle((unused, cause) -> {
                if (cause != null) {
                    logger.warn("Failed to stop the command executor:", cause);
                } else {
                    logger.info("Disabled replication");
                }
                return status();
            });
```